source/dotnet/Test/AdaptiveCards.Test/AdaptiveCardApiTests.cs

```diff
@@ -814,5 +814,176 @@ public void BackgroundImageBackCompatWithLegacyUriType()
+            var actionToggleVisibility = (AdaptiveToggleVisibilityAction)actionSet.Actions[3];
+            Assert.IsTrue(actionToggleVisibility.Title == "Toggle");
+            Assert.IsTrue(actionToggleVisibility.TargetElements.Count != 0);
+        }
+
+        [TestMethod]
+        public void TestObjectModelActionSetElement()
+        {
+            AdaptiveCard card = new AdaptiveCard(new AdaptiveSchemaVersion("1.2"));
+            AdaptiveActionSet actionSet = new AdaptiveActionSet();
+            card.Body.Add(actionSet);
+
+            AdaptiveSubmitAction submitAction = new AdaptiveSubmitAction
+            {
+                Title = "Action.Submit",
+                DataJson = "{\"x\": 13}"
+            };
+            actionSet.Actions.Add(submitAction);
+
+            AdaptiveOpenUrlAction openUrlAction = new AdaptiveOpenUrlAction
+            {
+                Title = "OpenUrl",
+                UrlString = "http://adaptivecards.io"
+            };
+            actionSet.Actions.Add(openUrlAction);
+
+            AdaptiveShowCardAction showCardAction = new AdaptiveShowCardAction
+            {
+                Title = "ShowCard",
+                Card = new AdaptiveCard
+                {
+                    Body = new List<AdaptiveElement>
+                    {
+                        new AdaptiveTextBlock
+                        {
+                            Text = "This is a show card"
+                        }
+                    }
+                }
+            };
+            actionSet.Actions.Add(showCardAction);
+
+            AdaptiveToggleVisibilityAction toggleVisibilityAction = new AdaptiveToggleVisibilityAction
+            {
+                Title = "Toggle",
+                TargetElements = new List<AdaptiveTargetElement> { "test" }
+            };
+            actionSet.Actions.Add(toggleVisibilityAction);
+
+            var expectedJson =
+@"{
+  ""type"": ""AdaptiveCard"",
+  ""version"": ""1.2"",
+  ""body"": [
+    {
+      ""type"": ""ActionSet"",
+      ""actions"": [
+        {
+          ""type"": ""Action.Submit"",
+          ""data"": {
+            ""x"": 13
+          },
+          ""title"": ""Action.Submit""
+        },
+        {
+          ""type"": ""Action.OpenUrl"",
+          ""url"": ""http://adaptivecards.io"",
+          ""title"": ""OpenUrl""
+        },
+        {
+          ""type"": ""Action.ShowCard"",
+          ""card"": {
+            ""type"": ""AdaptiveCard"",
+            ""version"": ""1.0"",
+            ""body"": [
+              {
+                ""type"": ""TextBlock"",
+                ""text"": ""This is a show card""
+              }
+            ]
+          },
+          ""title"": ""ShowCard""
+        },
+        {
+          ""type"": ""Action.ToggleVisibility"",
+          ""targetElements"": [
+            ""test""
+          ],
+          ""title"": ""Toggle""
+        }
+      ]
+    }
+  ]
+}";
```

Comment:
[nit] Indent this to line up with the code

Reply:
The indentation adds spaces that make the comparisson fail

Reply:
Makes sense. Could you add a comment to that effect?

---
In reply to: [273150076](https://github.com/Microsoft/AdaptiveCards/pull/2614#discussion_r273150076) [](ancestors = 273150076)